multipart_http_signer.go

```diff
@@ -0,0 +1,232 @@
+package tests
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	. "github.com/smartystreets/goconvey/convey"
+
+	"github.com/beyondstorage/go-storage/v4/pairs"
+	"github.com/beyondstorage/go-storage/v4/pkg/randbytes"
+	"github.com/beyondstorage/go-storage/v4/types"
+)
+
+func TestMultipartHTTPSignerCreateMultipart(t *testing.T, store types.Storager) {
+	Convey("Given a basic Storager", t, func() {
+		signer, ok := store.(types.MultipartHTTPSigner)
+		So(ok, ShouldBeTrue)
+
+		Convey("When CreateMultipart via QuerySignHTTPCreateMultipart", func() {
+			path := uuid.New().String()
+			req, err := signer.QuerySignHTTPCreateMultipart(path, time.Duration(time.Hour))
+
+			Convey("The error should be nil", func() {
+				So(err, ShouldBeNil)
+
+				So(req, ShouldNotBeNil)
+				So(req.URL, ShouldNotBeNil)
+			})
+
+			client := http.Client{}
+			_, err = client.Do(req)
+
+			Convey("The request returned error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("List with ModePart should get the object without error", func() {
+				it, err := store.List(path, pairs.WithListMode(types.ListModePart))
+
+				So(err, ShouldBeNil)
+
+				o, err := it.Next()
```

Comment:
How about recording the `MultipartID` and passing it as the deferred call's argument so that there's ni need to call `List` again?

Reply:
This is not feasible because tests at the same level as `list` do not run the `list` test case and end the test, so there is no `o`.

Reply:
How about use `store.Create()` to build the  `o *Object`?

Reply:
> How about use `store.Create()` to build the `o *Object`?

If we use `store.Create()` to build `o *Object` we still need `multipartID`.